http-client/src/main/java/io/airlift/http/client/HttpClientModule.java

```diff
@@ -203,6 +207,10 @@ public void destroy()
             if (pool != null) {
                 pool.close();
                 pool = null;
+                if(injector.getExistingBinding(Key.get(MBeanExporter.class)) != null) {
```

Comment:
This class is effectively a provider. Instead of doing the export directly, we should bind the JMX export when we bind IO pool and let the export happen the normal way.